File: frontend/console/src/api/signal.js
import Axios from "@/http";

export function createSignal(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSignal(signalId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/signal/${ signalId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSignal(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/signal', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignal(signalId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/signal/${ signalId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalWithFlag(signalId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/signal/${ signalId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSignals() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/signal/all').then(res => {
            resolve(res)
        })
    })
}

export function querySignal (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/signal.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SIGNAL = "/config/signal"
const URL_ROUTER_SIGNAL_BY_ID = "/config/signal/:signal-id"
const URL_ROUTER_SIGNAL_QUERY = "/config/signal/query"
const URL_ROUTER_SIGNAL_ALL = "/config/signal/all"

func SignalRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SIGNAL, config.CreateSignal)
	r.GET(URL_ROUTER_SIGNAL_BY_ID, config.RetrieveSignal)
	r.PUT(URL_ROUTER_SIGNAL, config.UpdateSignal)
	r.DELETE(URL_ROUTER_SIGNAL_BY_ID, config.DeleteSignal)
	r.PUT(URL_ROUTER_SIGNAL_BY_ID, config.DeleteSignalWithFlag)
	r.POST(URL_ROUTER_SIGNAL_QUERY, config.QuerySignal)
	r.GET(URL_ROUTER_SIGNAL_ALL, config.GetAllSignals)
	return r
}